src/documentation/doc-alias-policy.md

```diff
@@ -0,0 +1,35 @@
+doc alias policy
+================
+
+Rust's documentation supports adding aliases to any declaration (such as a
+function, type, or constant), using the syntax `#[doc(alias = "name")]`. We
+want to use doc aliases to help people find what they're looking for, while
+keeping those aliases maintainable and high-value. This policy outlines the
+cases where we add doc aliases, and the cases where we omit those aliases.
+
+- We must have a reasonable expectation that people might search for the term
+  in the documentation search. Rust's documentation provides a name search, not
+  a full-text search; as such, we expect that people may search for plausible
+  names, but that for more general documentation searches they'll turn to a web
+  search engine.
+  - Related: we don't expect that people are currently searching Rust
+    documentation for language-specific name from arbitrary languages they're
+    familiar with, and we don't want to add that as a new documentation search
+    feature; please don't add aliases based on your favorite language. Those
+    mappings should live in separate guides or references. We do expect that
+    people might look for the Rust name of a function they reasonably expect to
+    exist in Rust (e.g. a system function or a C library function), to try to
+    figure out what Rust called that function.
+- The proposed alias must be a name we would plausibly have used for the
+  declaration. For instance, `delete` for `remove`, or `popcnt` and `popcount`
+  for `count_ones`, or `umask` for `mode`, or `mkdir` for `create_dir`. This
+  feeds into the reasonable expectation that someone might search for the name
+  and expect to find it ("what did Rust call `mkdir`").
+- There must be an obvious single target for the alias that is an *exact*
```

Comment:
Am I reading it correctly that this would have disallowed https://github.com/rust-lang/rust/pull/79233. `malloc` doesn't have a single logical equivalent in Rust, instead we have several methods which cover the full range of functionality `malloc` provides.

Reply:
Also in the case of https://github.com/rust-lang/rust/pull/86596#issuecomment-871952116 (which I'm okay not being merged, not going to argue that here) there are two targets for an alias proposed: `windows` and `array_windows`.

Under the current wording of the guidance policy it reads to me that because there are const and a non-const variant of the API, we wouldn't be able to add an alias to either. I can see similar issues apply to async/non-async versions of similar APIs in the future as well.

I wonder if perhaps the phrasing on this point could be relaxed somewhat.

Reply:
Yes, and we discussed those malloc aliases in the meeting as examples of aliases we don't want to have. @Amanieu suggested that he may send a PR to remove them. (We might potentially point malloc to the corresponding function in the allocator API instead, if and only if there's a single exact match.)

Reply:
If we have a const and a non-const version of the same declaration, but they're really the same function, that seems like an implementation detail that wouldn't preclude an alias. I'll clarify that.